students/aaron/lesson02/assignment/src/charges_calc.py

```diff
@@ -0,0 +1,133 @@
+    log_file = datetime.datetime.now().strftime("%Y-%m-%d")+'.log'
+    formatter = logging.Formatter(log_format)
+    logger.setLevel(level)
+    file_level = logging.WARNING if level == logging.DEBUG else level
+    file_handler = logging.FileHandler(log_file)
+    file_handler.setLevel(file_level)
+    file_handler.setFormatter(formatter)
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(level)
+    console_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+    logger.addHandler(console_handler)
+    return logger
+
+def parse_cmd_arguments():
+    """
+    Parses cli arguments
+    """
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d',
+                        '--debug',
+                        help='Debug level: 0 (quietest) to 3 (most verbose)',
+                        required=False)
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """
+    Loads a json file
+    """
+    LOGGER.debug("load_rentals_file()")
+    try:
+        with open(filename) as file:
+            LOGGER.debug("Loading JSON data from %s", filename)
+            data = json.load(file)
+            LOGGER.debug("Successfully loaded JSON data.")
+    except Exception as err:
+        LOGGER.error("Failed to load JSON data. %s", err)
+        raise err
+    return data
+
+def calculate_additional_fields(data):
+    """
+    Calculates additional fields
+    """
+    keys = ['rental_start', 'rental_end', 'price_per_day', 'units_rented']
+    for value in data.values():
+        missing_keys = [x for x in keys if not x in value.keys()]
+        if missing_keys:
+            LOGGER.warning("Could not find required key(s) %s in %s.  Skip.",
+                           missing_keys,
+                           value)
+            continue
+        try:
+            rental_start = datetime.datetime.strptime(value['rental_start'],
+                                                      '%m/%d/%y')
+            rental_end = datetime.datetime.strptime(value['rental_end'],
+                                                    '%m/%d/%y')
+        except ValueError as err:
+            LOGGER.warning("Failed to calculate time from %s or %s. %s",
+                           value['rental_start'],
+                           value['rental_end'],
+                           err)
+        if value['units_rented'] == 0:
+            LOGGER.error("Units rented is 0 for %s.  Skip.", value)
+            continue
+        if rental_start >= rental_end:
+            LOGGER.warning("Rental start %s on or after end %s in %s.  Skip.",
+                           rental_start,
+                           rental_end,
+                           value)
+            continue
```

Comment:
What is purpose of continue here?

Reply:
The continue will skip to the next iteration of the loop without updating the data.  The assumption is that we were given bad data in this entry, so by the GIGO philosophy instead of doing any further processing on the entry, we are going to leave it as is and move on to the next entry.